File: offgame-core/src/main/java/org/retal/offgame/dto/ResourcesDetails.java
package org.retal.offgame.dto;

import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResourcesDetails {

    private String planetName;
    private List<ResourceDetail> resourceDetails;
    private ResourcesDTO totalResources;
}

File: offgame-ui/src/app/model/resource/ResourcesDetails.ts
import {Resources} from "./Resources";
import {ResourceDetail} from "./ResourceDetail";

export class ResourcesDetails {

    planetName!: string
    resourceDetails: ResourceDetail[] = []
    totalResources: Resources

    constructor(data: any) {
        Object.assign(this, data);

        this.resourceDetails = data.resourceDetails.map((object: any) => new ResourceDetail(object))

        this.totalResources = new Resources(data.totalResources)
        this.totalResources.initResourceLogic()
    }
}